api/groups/urlParams.go

```diff
@@ -38,12 +39,39 @@ func parseAccountQueryOptions(c *gin.Context) (common.AccountQueryOptions, error
 		return common.AccountQueryOptions{}, err
 	}
 
-	onStartOfEpoch, err := parseUintUrlParam(c, common.UrlParameterOnStartOfEpoch)
+	onStartOfEpoch, err := parseUint32UrlParam(c, common.UrlParameterOnStartOfEpoch)
 	if err != nil {
 		return common.AccountQueryOptions{}, err
 	}
 
-	options := common.AccountQueryOptions{OnFinalBlock: onFinalBlock, OnStartOfEpoch: onStartOfEpoch}
+	blockNonce, err := parseUint64UrlParam(c, common.UrlParameterBlockNonce)
+	if err != nil {
+		return common.AccountQueryOptions{}, err
+	}
+
+	blockHash := parseStringUrlParam(c, common.UrlParameterBlockHash)
+	if err != nil {
+		return common.AccountQueryOptions{}, err
+	}
+
+	blockRootHash := parseStringUrlParam(c, common.UrlParameterBlockRootHash)
```

Comment:
Should we parse as hex, and return an error if the string is invalid? Or let the Node alone do the validation?

Reply:
better parse the hex on proxy level, as to avoid an additional call to the node(s)

Reply:
Fixed.